capstone-backend/src/main/java/com/google/sps/servlets/ClubServlet.java

```diff
@@ -1,166 +1,134 @@
-    clubEntity.setProperty("ownerID", ownerID);
-    clubEntity.setProperty("inviteIDs", inviteIDs);
-    clubEntity.setProperty("memberIDs", new ArrayList<Long>());
-    clubEntity.setProperty("gbookID", gbookID);
-
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    datastore.put(clubEntity);
+    Club club = new Club(id, name, description, ownerID, gbookID);
+    ApiFuture<WriteResult> future = clubs.document(id).set(club);
+    try {
+      System.out.println("Update time : " + future.get().getUpdateTime());
+    } catch (Exception e) {
+      System.err.println("Error: " + e);
+    }
+    Gson gson = new Gson();
+    response.setContentType("application/json;");
+    response.getWriter().println(gson.toJson(club));
   }
 
   @Override
   public void doPut(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    JsonObject jsonObject = createRequestBodyJson(request);
-    long id = jsonObject.get("id").getAsLong();
-    Entity clubEntity = new Entity("Club", id);
-
-    String name = (String) clubEntity.getProperty("name");
-    String description = (String) clubEntity.getProperty("description");
-    String announcement = (String) clubEntity.getProperty("announcement");
-    long ownerID = (long) clubEntity.getProperty("ownerID");
-    String gbookID = (String) clubEntity.getProperty("gbookID");
+    JsonObject jsonObject = Utility.createRequestBodyJson(request);
+    String id = jsonObject.get("id").getAsString();
+    Map<String, Object> update = new HashMap<>();
 
     if (jsonObject.get("name") != null) {
-      name = jsonObject.get("name").getAsString();
+      update.put("name", jsonObject.get("name").getAsString());
     }
     if (jsonObject.get("description") != null) {
-      description = jsonObject.get("description").getAsString();
-    }
-    if (jsonObject.get("announcement") != null) {
-      announcement = jsonObject.get("announcement").getAsString();
-    }
-    if (jsonObject.get("ownerID") != null) {
-      ownerID = jsonObject.get("ownerID").getAsLong();
+      update.put("description", jsonObject.get("description").getAsInt());
     }
     if (jsonObject.get("gbookID") != null) {
-      gbookID = jsonObject.get("gbookID").getAsString();
+      update.put("gbookID", jsonObject.get("gbookID").getAsString());
     }
 
-    clubEntity.setProperty("name", name);
-    clubEntity.setProperty("description", description);
-    clubEntity.setProperty("announcement", announcement);
-    clubEntity.setProperty("ownerID", ownerID);
-    clubEntity.setProperty("gbookID", gbookID);
-
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    datastore.put(clubEntity);
+    ApiFuture<WriteResult> writeResult = clubs.document(id).set(update, SetOptions.merge());
+    try {
+      System.out.println("Update time : " + writeResult.get().getUpdateTime());
+    } catch (Exception e) {
+      System.err.println("Error: " + e);
```

Comment:
Same here: probably need some additional error-handling logic. You might also need to send the updated document in the response payload to handle updating the front-end without a refresh, though you could also do optimistic front-end updates and notify the user when changes aren't saved.

Reply:
That's a good idea! I'll add that.

Reply:
I did that in my Post, but forgot to include in my Put, will add.